lib/ssh_scan/crypto.rb

```diff
@@ -5,21 +5,37 @@ module Crypto
     class PublicKey
       def initialize(key)
         @key = key
-        @supported = check_supported
+        @rsa_supported = check_rsa_supported
+        @dsa_supported = check_dsa_supported
         if @key.is_a?(OpenSSL::PKey::RSA)
           @data_string = OpenSSL::ASN1::Sequence([
            OpenSSL::ASN1::Integer.new(@key.public_key.n),
            OpenSSL::ASN1::Integer.new(@key.public_key.e)
           ])
         end
+        if @key.is_a?(OpenSSL::PKey::DSA)
+          @data_string = OpenSSL::ASN1::Sequence([
+           OpenSSL::ASN1::Integer.new(@key.public_key.p),
+           OpenSSL::ASN1::Integer.new(@key.public_key.q),
+           OpenSSL::ASN1::Integer.new(@key.public_key.g)
+          ])
+        end
       end
 
-      def check_supported
+      def check_rsa_supported
```

Comment:
in ruby, if you're returning binary result, it's customary to make it a question.

How about "rsa_supported?" then you don't need to keep instance variables with the result.